docs/design-documents/20241024-conduit-cli.md

```diff
@@ -0,0 +1,375 @@
+we should prompt to set up a working Conduit environment via `conduit init`.
+
+It does not require having Conduit running.
+
+A source and a destination could be provided using the same connectors reference as described
+[here](https://conduit.io/docs/using/connectors/referencing).
+
+#### Arguments
+
+| Name | Description | Required | Default Value |
+|------|-------------|----------|---------------|
+| name  |  Pipeline file name and pipeline name  | No | `pipeline-#` (`pipeline-#.yaml`) |
+
+#### Flags
+
+| Name | Description | Required | Default Value |
+|------|-------------|----------|---------------|
+| destination  |  Plugin name of the destination connector  | No | `log` |
+| pipelines.path  |  Where to initialize a new pipeline | No | `.` (current directory) |
+| source  |  Plugin name of the source connector  | No | `generator` |
+
+#### `--help`
+
+```bash
+conduit pipelines init
+conduit pipelines init my-first-pipeline
+conduit pipelines init my-first-pipeline --pipelines.path ~/my-other-path
+conduit pipelines init --source file@v1.0 --destination file
+```
+
+### `conduit pipelines ls`
+
+This command will list the running pipelines.
+
+It requires having Conduit previously running.
+
+#### Flags
+
+| Name | Description | Required | Default Value |
+|------|-------------|----------|---------------|
+| grpc.address | address for serving the gRPC API | No | ":8084" |
+
+#### `--help`
+
+```bash
+$ conduit pipelines ls
+NAME            STATUS
+my-pipeline     running
+my-other        degraded      
+```
+
+### `conduit pipelines describe ID`
+
+This command will describe the topology of the pipeline.
+
+It requires having Conduit previously running.
+
+It requires the pipeline id as argument.
+
+#### Arguments
+
+| Name | Description | Required | Default Value |
+|------|-------------|----------|---------------|
+| id  |  pipeline id to describe | Yes | |
+| grpc.address | address for serving the gRPC API | No | ":8084" |
+
+#### `--help`
+
+```bash
+$ conduit pipelines describe [ID]
+
+EXAMPLE:
+
+$ conduit pipelines describe my-pipeline
+Source: kafka
+  Processor: avro.encode
+Destination: kafka
+```
+
+### `conduit connectors ls`
+
+This command will list all the available connectors.
+
+It requires having Conduit previously running.
+
+#### `--help`
+
+```bash
+$ conduit connectors ls
+ID                PLUGIN                          TYPE         PIPELINE
+my-source         postgres@v0.2.0                  builtin      file-to-postgres
+my-destination    conduit-connector-http@0.1.0.   standalone   my-other-pipeline
+```
+
+### `conduit connectors describe [--plugin] [--id]`
+
+This command will describe the connector configuration available.
+
+When using `--plugin` will describe the configuration for that connector plugin.
+
+When using `--id` will require having Conduit previously running since it'll describe the existing running connector.
```

Comment:
So these 2 are mutually exclusive?

Reply:
@hariso correct

Reply:
I'm not sure I like conflating these two things in a single command 😕

Reply:
How about `conduit connectors plugins ...` for accessing plugins? 🤔 Or is this too verbose?

Reply:
If having mutually exclusive flags is an issue, then I think we could have the separation of `connectors` and `connector-plugins`. I'm not sure having `plugins` as a subcommand of `connectors` helped with discoverability.

Reply:
I too think it would be good if we find something that doesn't combine these two things, since one is about describing a connector plugin, and the other one is about a running connector.

How about we have `get-specs` and `describe`?

Reply:
Maybe it's not too bad after all. That could open the possibility of listing all conector and processor plugins:

```bash
$ conduit connector-plugins ls
$ conduit processor-plugins ls
```

Reply:
Would we then move (un)install under this? Looks good to me. :+1:

Reply:
@lovromazgon @hariso just updated it for connectors. Let me know what you think

Reply:
We might as well alias it under `conduit connectors plugins` and provide both options, then we have a subcommand that is directly discoverable, while also having the one that makes sense hierarchically 🤷  WDYT?

Reply:
@hariso 

> Would we then move (un)install under this? Looks good to me. 👍

Not planning to include `install` or `uninstall` as part of this release, but you're right. It'd be under those commands.